scripts/glutton/update/src/main.rs

```diff
@@ -0,0 +1,176 @@
+use crate::glutton_para::runtime_types::sp_arithmetic::per_things::Perbill;
+use clap::{Parser, ValueEnum};
+use subxt::{
+	dynamic::Value,
+	ext::sp_core::{sr25519::Pair as Sr25519Pair, Pair},
+	tx::PairSigner,
+	utils::AccountId32,
+	OnlineClient,
+};
+
+mod config;
+use crate::config::GluttonConfig;
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord, ValueEnum)]
+enum RelayChain {
+	Local,
+	Kusama,
+	Versi,
+}
+
+#[derive(Parser, Debug)]
+#[command(author, version, about, long_about = None)]
+struct Cli {
+	/// Relay chain context selection: kusama, versi or local
+	#[clap(value_enum)]
+	#[arg(short, long)]
+	relay: RelayChain,
+
+	/// Update Glutton Parachains from 'para_id = from' (inclusive)
+	#[arg(short, long)]
+	from: u32,
+
+	/// Update Glutton Parachains to 'para_id = to' (inclusive)
+	#[arg(short, long)]
+	to: u32,
+
+	/// Optional new 'storage' value to update
+	/// Value in parts per billion. E.g: 500000000 equals 50% of a block `ref_time` consumed
+	#[arg(short, long)]
+	storage: Option<u32>,
+
+	/// Optional new 'compute' value to update
+	/// Value in parts per billion. E.g: 500000000 equals 50% of a block `proof_size` consumed
+	#[arg(short, long)]
+	compute: Option<u32>,
+
+	/// Sudo account secret seed in hex format
+	#[arg(short, long)]
+	private_key: String,
+}
+
+// Generate an interface that we can use from the node's metadata.
+#[subxt::subxt(runtime_metadata_path = "./artifacts/glutton_metadata.scale")]
+pub mod glutton_para {}
+
+type RuntimeCall = glutton_para::runtime_types::glutton_runtime::RuntimeCall;
+type GluttonCall = glutton_para::runtime_types::pallet_glutton::pallet::Call;
+
+async fn get_account_nonce(
+	client: &OnlineClient<GluttonConfig>,
+	account: &AccountId32,
+) -> Result<u32, Box<dyn std::error::Error>> {
+	let runtime_api_call = subxt::dynamic::runtime_api_call(
+		"AccountNonceApi",
+		"account_nonce",
+		vec![Value::from_bytes(account)],
+	);
+
+	let nonce = client
+		.runtime_api()
+		.at_latest()
+		.await?
+		.call(runtime_api_call)
+		.await?
+		.to_value()?
+		.as_u128()
+		.unwrap() as u32;
+
+	Ok(nonce)
+}
+
+async fn update_glutton(
+	client: &OnlineClient<GluttonConfig>,
+	call: RuntimeCall,
+	account: &AccountId32,
+	signer: &PairSigner<GluttonConfig, Sr25519Pair>,
+) -> Result<(), Box<dyn std::error::Error>> {
+	// Get signer's `nonce`
+	let nonce = get_account_nonce(client, account).await?;
+
+	// Build the tx
+	let tx = glutton_para::tx().sudo().sudo(call);
+
+	// Build the signed tx
+	let signed_tx = client
+		.tx()
+		.create_signed_with_nonce(&tx, signer, nonce, Default::default())
+		.unwrap();
+
+	// Submit and watch the tx
+	let in_block = signed_tx.submit_and_watch().await.unwrap().wait_for_in_block().await.unwrap();
```

Comment:
Just to note that `.wait_for_in_block()` doesn't guarantee that the block will be finalized; use `.wait_for_finalized()` for that, or even `.wait_for_finalized_success()` to also verify that the tx was successfully executed (ie an ExtrinsicSuccess event was produced from its inclusion)